substrate/frame/support/src/traits/try_runtime/mod.rs

```diff
@@ -161,22 +161,31 @@ impl<BlockNumber: Clone + sp_std::fmt::Debug + AtLeast32BitUnsigned> TryState<Bl
 		match targets {
 			Select::None => Ok(()),
 			Select::All => {
-				let mut error_count = 0;
+				let mut errors = Vec::<TryRuntimeError>::new();
+
 				for_tuples!(#(
-					if let Err(_) = Tuple::try_state(n.clone(), targets.clone()) {
-						error_count += 1;
+					if let Err(err) = Tuple::try_state(n.clone(), targets.clone()) {
+						errors.push(err);
```

Comment:
why not just log the error here?

Reply:
There are a lot of intermediate logs from migrations. I think it makes sense to have a comprehensive error report at the end that shows all the errors at once instead of needing to scroll.

Reply:
Anything is good to me.

Ideally try_state implementations should just log its error with precise file:line information and just return an Error with maybe the pallet name. But currently there is a lot of use of `ensure!(..., "error message")` and this just goes inside the error.